Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -243,6 +243,107 @@ theorem mem_oneCocycles_iff (f : G → A) :
+    (hg : ∀ n : ℕ, f (g ^ n) = ∑ i in Finset.range n, A.ρ (g ^ i) (f g)) :
+    f ∈ oneCocycles A := (mem_oneCocycles_iff f).2 fun x y => by
+  rcases hgmem x with ⟨m, rfl⟩
+  rcases hgmem y with ⟨n, rfl⟩
+  simp only [← pow_add, hg, map_pow, map_sum, add_comm ((Finset.range n).sum _)]
+  convert Finset.sum_range_add _ _ _ using 3
+  simp only [pow_add, LinearMap.mul_apply]
+
+variable (A)
+
+/-- The `k`-linear map sending `x : A` to `∑ ρ_A(gⁱ)(x)` for `0 ≤ i < n.` Used for defining
+1-cocycles when `G` is finite and cyclic. -/
+noncomputable def sumρPowers (g : G) (n : ℕ) : A →ₗ[k] A :=
+∑ i in Finset.range n, A.ρ (g ^ i)
+
+variable {A}
+
+theorem sumρPowers_def (g : G) (n : ℕ) :
+    sumρPowers A g n = ∑ i in Finset.range n, A.ρ (g ^ i) := rfl
+
+@[simp] theorem sumρPowers_zero (g : G) : sumρPowers A g 0 = 0 := by
+  simp only [sumρPowers_def, Finset.range_zero, map_pow, Finset.sum_empty]
+
+@[simp] theorem sumρPowers_one (g : G) : sumρPowers A g 1 = 1 := by
+  simp only [sumρPowers_def, Finset.range_one, map_pow, Finset.sum_singleton, pow_zero]
+
+theorem sumρPowers_orderOf_eq_norm [Fintype G] (g : G) (hg : ∀ h, h ∈ Submonoid.powers g) :
+    sumρPowers A g (orderOf g) = (norm A).hom := by
+  ext
+  simp only [sumρPowers, norm, map_pow, LinearMap.coeFn_sum, Finset.sum_apply, Rep.homMk_hom]
+  exact Finset.sum_bij (fun n _ => g ^ n) (fun a ha => Finset.mem_univ _)
+    (fun a _ => by simp only [map_pow])
+    (fun a b ha hb (hab : g ^ a = g ^ b) => Nat.ModEq.eq_of_lt_of_lt (pow_eq_pow_iff_modEq.1 hab)
+      (Finset.mem_range.1 ha) (Finset.mem_range.1 hb))
+    (fun b _ => by
+    { rcases hg b with ⟨i, hi⟩
+      exact ⟨i % orderOf g, Finset.mem_range.2 (Nat.mod_lt i $ IsOfFinOrder.orderOf_pos
+        (isOfFinOrder_of_finite g)), by simp only [Finset.mem_univ, pow_mod_orderOf, hi] at *⟩ })
+
+theorem sumρPowers_add (g : G) (m n : ℕ) :
+    sumρPowers A g (m + n) = sumρPowers A g m + (A.ρ g) ^ m * sumρPowers A g n := by
+  simp only [sumρPowers_def, map_pow, Finset.sum_range_add, pow_add, add_right_inj,
+    ← Finset.mul_sum]
+
+theorem sumρPowers_orderOf_mul (g : G) (i : ℕ) :
+    sumρPowers A g (orderOf g * i) = i * sumρPowers A g (orderOf g) := by
+  induction i with
+  | zero =>
+    simp only [Nat.zero_eq, mul_zero, sumρPowers_def, Finset.range_zero, map_pow,
+      Finset.sum_empty, Nat.cast_zero, zero_mul]
+  | succ i hi =>
+    simp only [sumρPowers_def] at *
+    simp only [Nat.cast_succ, mul_add, add_mul, ← hi, one_mul, mul_one, Nat.succ_eq_add_one,
+      Finset.sum_range_add, pow_add, pow_mul, pow_orderOf_eq_one g, one_pow]
+
+/-- Given a generator `g` of a finite cyclic group `G`, a representation `A` of `G`, and an element
+`x : A` of norm zero, the map `G → A` sending `gⁱ ↦ ∑ ρ_A(gʲ)(x)` for `0 ≤ j < i` is
+a 1-cocycle. -/
+noncomputable def oneCocyclesOfGenerator [Fintype G] (x : A) (g : G)
+    (hg : ∀ h, h ∈ Submonoid.powers g) (hx : (norm A).hom x = 0) :
```

Comment:
We already have a typeclass `IsCyclic G`, but I am wondering whether we could introduce a typeclass `IsCyclicGenerator g` which would mean `∀ x, x ∈ zpowers g`.